### ISSUE TITLE
Enhance Bulk Download

### ISSUE THREAD

**Issue by w4st3dtim3** [HuiiBuh/InstagramDownloader#110]
Hi there,

first of all thank you for the addon. Its very helpful.

I have two questions regarding the bulk download. I always use it on the "Saved" photos.

It would be immensely helpful, if it would be possible to set up how many posts should be included in the bulk download. Eg, now I can just press stop and cant really control how many items are downloaded. But if I could chose to say I want to download specific numbers auf x items/posts it would be brilliant.

The other thing that would me immensely helpful, is that in bulk download not the file names of the Instagram account is included like when I download a single file. This makes organizing a pain. Maybe its possible to use the same file names like in single picture download.

Thanks.

Best regards.

**Comment by HuiiBuh** [comment]
So you want: 
1. Limit download all to a specific number of posts
2. Include the account name in the downloaded image files

I think this will take some time. I just created a major release and will slow down a bit after this large release. If you want to create a PR I would be more than willing to merge it

**Comment by w4st3dtim3** [comment]
Hi,

yes correct! That would be awesome. No worries. Take your time. Looking forward to these features.

**Comment by nmkeraj** [comment]
Hi, After so many trials when I could download videos once it doesn't work. It stops after collecting. I get "Please wait until the download continues in the background" and nothing happens. I use Chrome and Firefox

**Comment by orazdow** [comment]
I agree having a generated filename that includes the post href or part of the image elemen's "alt" property if it exits would be good that the random guid filenames. I tried looking through the source to try and change that but got lost figuring out the bulkDownloader's functions. I thought the message object in downloadContent() might have been the thing to modify but wasn't sure, and then it would have to be a function of things obtained from the post elements.

**Comment by HuiiBuh** [comment]
The filename gets created at the very end of the download.ts file in the background folder.
I think the alt property is not really. Useful, because it just something like _image with people_.
There is a discussion about file naming, perhaps this helps you...

**Comment by topotech** [comment]
@HuiiBuh I may try to create a possible solution to two of these problems. This is defined in` downloadBulk()` from `background/download.ts`

**Comment by HuiiBuh** [comment]
Yes the download happens in this function. But you may have to add some code in other places...
